mantidproject/mantid
# Fix sliceviewer view tests that are failing locally

**Description of work.**
Fixes Sliceviewer Tests. These were consistently failing locally on two different machines by running unit test directly rather than using ctest.
Additionally, this should fix some of the flakey-ness we've seen with these tests too.

These changes can be surmised into the following:
- Chunking out presenter and model tests from the view
- Mocking as much as possible, especially avoiding testing workspace and their methods (these are unit tested in the Mantid framework), and relying on mocks instead
- View tests which directly tested MPL code were left as-is 
- Removed one test that negative log scale aren't allowed in MPL. (This is documented on their side so we'll just assume MPL have tests to check this)

**To test:**
- Code review
- Run tests locally using pytest / unit test (not ctest), check they pass:
`<build_dir>/bin/mantidpython --classic -m pytest <mantid_src>/qt/python/mantidqt/mantidqt/widgets/sliceviewer/test/`

Part of #33038 - This was blocking us from running tests locally

*This does not require release notes* because it should not introduce any user facing changes



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
